packages/medusa/src/types/note.ts

```diff
@@ -0,0 +1,26 @@
+import { Note } from "../models/note"
+import { User } from "../models/user"
+import { PartialPick } from "./common"
+
+export interface CreateNoteInput {
+  value: string
+  resource_type: string
+  resource_id: string
+  author_id: string
```

Comment:
This should be optional in the case where the note is being created by a "system" e.g. a subscriber that calls the service layer directly.

Reply:
I take the blame for this one. I was too quick on the keyboard in the previous review when I saw the optional API payload properties 👍  Nice one!